File: packages/python-sdk/payxcommerce/util/redactor.py
from __future__ import annotations

import re
from typing import Any

_PATTERN = re.compile(r"(secret|token|signature|authorization|password|key|client_secret|secret_key|webhook_secret)([^\s:=]*)?([:=]\s*)?([A-Za-z0-9_\-.!@$%^&*+/=]+)", re.IGNORECASE)


def text(message: str) -> str:
    return _PATTERN.sub(r"\1\2\3[redacted]", message)


def context(values: dict[str, Any]) -> dict[str, Any]:
    redacted: dict[str, Any] = {}
    for key, value in values.items():
        if re.search(r"secret|token|signature|authorization|password|key", str(key), re.IGNORECASE):
            redacted[key] = "[redacted]"
        elif isinstance(value, str):
            redacted[key] = text(value)
        elif isinstance(value, dict):
            redacted[key] = context(value)
        else:
            redacted[key] = value
    return redacted

File: packages/node-sdk/src/util/redactor.js
'use strict';

function text(message) {
  return String(message).replace(/(secret|token|signature|authorization|password|key|client_secret|secret_key|webhook_secret)([^\s:=]*)?([:=]\s*)?([A-Za-z0-9_\-.!@$%^&*+/=]+)/gi, '$1$2$3[redacted]');
}

function context(values) {
  const redacted = {};
  for (const [key, value] of Object.entries(values || {})) {
    if (/secret|token|signature|authorization|password|key/i.test(key)) {
      redacted[key] = '[redacted]';
    } else if (typeof value === 'string') {
      redacted[key] = text(value);
    } else if (value && typeof value === 'object' && !Array.isArray(value)) {
      redacted[key] = context(value);
    } else {
      redacted[key] = value;
    }
  }
  return redacted;
}

module.exports = { text, context };
